### ISSUE TITLE
Refactor workflow hydration process

### ISSUE THREAD

**Issue by mostaphaRoudsari** [ladybug-tools/queenbee#32]
Hydrating a workflow is the process of replacing referenced workflow variables by the input values. We already have a method that hydrates the workflow. It does the job but it is currently written to both validate the variables and get the values while doing the hydration. These are technically 3 different tasks.

1. The formatting for variables should be validated when the value is assigned to the object.
2. Retrieving the values for each argument should be implemented in the object.
3. Replacing the workflow variables with their respective value which should be done during the hydration.

Basically the workflow and variables should be validated well before the hydration happens. This will save us from validation issues during the translation of queenbee workflows to other platforms.

**Comment by AntoineDao** [comment]
Righto, as discussed today with @mostaphaRoudsari I will tackle this issue this week. Good way to get back into the flow of things.

**Comment by mostaphaRoudsari** [comment]
@AntoineDao, items 1 and 2 are already implemented. We still need item 3.
